src/routes/(picsure)/(public)/explorer/genome-filter/+page.svelte

```diff
@@ -0,0 +1,129 @@
+
+  import { VariantKeys } from '$lib/models/InfoColumns';
+  import { Option } from '$lib/models/GemoneFilter';
+  import GeneFilterStore from '$lib/stores/GenomicFilter';
+  import FilterStore from '$lib/stores/Filter';
+  import InfoColumns from '$lib/stores/InfoColumns';
+  let {
+    selectedGenes,
+    selectedFrequency,
+    selectedConsequence,
+    snpSearch,
+    snpConstraint,
+    generateFilter,
+    clearGeneFilters,
+  } = GeneFilterStore;
+  let { addFilter } = FilterStore;
+  let { error: infoColumnError, loadInfoColumns, getInfoColumn } = InfoColumns;
+
+  let selectedOption: Option = Option.Name;
+
+  function onComplete() {
+    const filter = generateFilter(selectedOption);
+    addFilter(filter);
+  }
+
+  onMount(async () => {
+    await loadInfoColumns();
+    if ($infoColumnError) {
+      toastStore.trigger({
+        message: $infoColumnError,
+        background: 'variant-filled-error',
+      });
+    }
+  });
+
+  $: canComplete =
+    (selectedOption === Option.Name &&
+      ($selectedGenes.length > 0 ||
+        $selectedFrequency.length > 0 ||
+        $selectedConsequence.length > 0)) ||
+    (selectedOption === Option.SNP && $snpSearch !== '' && $snpConstraint !== '');
+</script>
+
+<svelte:head>
+  <title>{branding.applicationName} | Gemonic Filter</title>
+</svelte:head>
+
+<Content>
+  <AngleButton href="/explorer">Back to Cohort builder</AngleButton>
+  <h2 class="text-center mb-4">Genomic Filtering</h2>
+  <Stepper
+    buttonCompleteLabel="Apply Filter"
+    on:complete={onComplete}
+    bind:canComplete
+    class="mx-5 p-4 border rounded-md border-surface-500-400-token"
+  >
+    <Step title="Select Filter Type">
+      <FilterType
+        bind:selectedOption
+        on:update={(event) => (selectedOption = event.detail.option)}
+      />
+    </Step>
+    <Step
+      title={selectedOption === Option.Name ? 'Gene with Variant' : 'Specific SNP'}
+      locked={!canComplete}
+    >
+      {#if selectedOption === Option.Name}
+        <div class="flex gap-3">
+          <Panel title="Select genes of interest" class="w-2/3">
+            <svelte:fragment slot="help">
+              <span class="text-error-500-400-token mr-2">Required</span>
+              <HelpInfoPopup id="gene-help-popup" text={getInfoColumn(VariantKeys.gene)} />
+            </svelte:fragment>
+            <SelectGenes />
+          </Panel>
+          <Panel title="Select variant frequency" class="w-1/3">
+            <svelte:fragment slot="help">
+              <HelpInfoPopup id="freq-help-popup" text={getInfoColumn(VariantKeys.frequency)} />
+            </svelte:fragment>
+            <SelectFrequency />
+          </Panel>
+        </div>
+        <div class="flex gap-3">
+          <Panel title="Variant consequence calculated" class="w-1/2">
+            <svelte:fragment slot="help">
+              {#if getInfoColumn(VariantKeys.consequence) && getInfoColumn(VariantKeys.severity)}
```

Comment:
Why just this one if statement for help info?

Reply:
Yes, this is because the statements are combined. If we're missing one then we'd have a random info bubble that says 'Severity' only.